magnolia/shared/src/main/scala-dotty/zio/config/magnolia/Descriptor.scala

```diff
@@ -0,0 +1,247 @@
+package zio.config.magnolia
+
```

Comment:
A few code refactoring such as separate functions for `product` and `sum` is reverted due to dotty errors.